### PR TITLE
Add suggest anomaly detector action to discover page

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description

This PR adds a `suggest anomaly detector` button to the discover page, when users click the button, a flyout pops up, and the index pattern extracted from the discover page is sent to a backend API which will return some suggested parameters from LLM, the suggested parameters contain features list and a optional category field used for creating anomaly detector, users can click the `create detector` button to create a new detector after they confirm the suggested parameters or update that if need.

The code change include:

- Register a new `DiscoverAction` into `Discover`, the register function is provided by OSD core, in this PR:https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7793
- Implement a new flyout for generating anomaly detector, we're not able to use the existing feature anywhere flyout because there're are many difference between the UI and existing feature anywhere flyout is associated with a visualization which is different from our case.
- Add a new node API `_generate_parameters` which calls some backend APIs in the ml-commons plugin.
- Some common API calls in the frontend use relative path, for example, `..${BASE_NODE_API_PATH}/_mappings`, which is not available for this feature, because the URI of the discover page is `app/data-explorer/discover#`, different from the URI in anomaly detection plugin: `app/anomaly-detection-dashboards#`, so I changed it to `${BASE_NODE_API_PATH}/_mappings` to avoid 404 error.

Some screenshots:

1. Entry point

image

2. Flyout
image

A demo video:

https://github.com/user-attachments/assets/e661f9d9-1baa-46d2-a101-5271cd89233d

### Issues Resolved

https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/issues/816

### Check List

- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaobinlong** [comment]
The PR depends on the [PR](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7793) in core which is still in reviewing, I'll add more tests for this PR during the reviewing process, @ohltyler @kaituo @jackiehanyang @owaiskazi19 @joshpalis @SuZhou-Joe @ruanyl could you help to review this PR? Thank you!

File: public/components/DiscoverAction/GenerateAnomalyDetector.tsx
Comment (jackiehanyang) [inline public/components/DiscoverAction/GenerateAnomalyDetector.tsx:102]: it seems like `fields` could be undefined or null, let's add a default value here to ensure reduce function won't operate on undefined or null

File: public/components/DiscoverAction/GenerateAnomalyDetector.tsx
Comment (jackiehanyang) [inline public/components/DiscoverAction/GenerateAnomalyDetector.tsx:381]: Please sync with header redesign team and the UX designer to make sure this is aligned with the header redesign project
Cc @AMoo-Miki @kgcreative

**Review by jackiehanyang — COMMENTED** [review]
Thanks for the change! This is the first pass of the review, raised some minor issues, will review it again once there are more tests. 

My main concern here is that the flyout code is exactly the same as the one used in the feature anywhere detector creation flyout. Can we reuse the feature anywhere detector creation flyout here instead of copying the code into a different folder? This would help avoid adding duplicate code and the need to maintain two sets of code

File: opensearch_dashboards.json
Comment (gaobinlong) [reply opensearch_dashboards.json:8]: I've changed that, thank you!

**Comment by gaobinlong** [comment]
> My main concern here is that the flyout code is exactly the same as the one used in the feature anywhere detector creation flyout. Can we reuse the feature anywhere detector creation flyout here instead of copying the code into a different folder? This would help avoid adding duplicate code and the need to maintain two sets of code

@jackiehanyang Thanks for your review, I meant to leverage the existing feature anywhere flyout, but failed to do that because there're many difference between this case and the feature anywhere:

1. Feature anywhere flyout is associated with a visualization which is used to extract the key information for creating anomaly detector, but for the case in this PR, information is extracted from the Discover page, no visualization exists, so it's hard to use the `AddAnomalyDetector` component directly in this case.
2. The UI is also different from feature anywhere, `Associate existing detector` and `show visualization` is not needed in this case, and a different `Categorical field` and a new `Timestamp field` are added to `Advanced options`:
image 
vs 
image

Pasted Graphic
vs
image

**Comment by xluo-aws** [comment]
Thanks, Jackie. Can we get one more maintainer help review this PR? We plan to release it in 2.17.

File: public/pages/ConfigureModel/components/FeatureAccordion/FeatureAccordion.tsx
Comment (jackiehanyang) [inline public/pages/ConfigureModel/components/FeatureAccordion/FeatureAccordion.tsx:116]: what's the motivation of adding this? I believe currently it's showing trash icon

File: public/components/DiscoverAction/SuggestAnomalyDetector.tsx
Comment (jackiehanyang) [inline public/components/DiscoverAction/SuggestAnomalyDetector.tsx:420]: How are you populating the detector name? Based on the screenshot, it looks like the detector name doesn't follow any kind of UUID naming convention. We don't allow duplicated detector names, how is this being handled here? 

By the way, the demo recording video is broken in the PR description. Please consider reuploading it

File: public/components/DiscoverAction/GenerateAnomalyDetector.tsx
Comment (jackiehanyang) [reply public/components/DiscoverAction/GenerateAnomalyDetector.tsx]: Did we have a tech writing review on the naming conventions? It might be a nitpick, but `Suggested anomaly detector` sounds more natural and implies that the detector is being recommended or proposed based on certain criteria, imo

File: public/pages/ConfigureModel/components/FeatureAccordion/FeatureAccordion.tsx
Comment (gaobinlong) [reply public/pages/ConfigureModel/components/FeatureAccordion/FeatureAccordion.tsx:116]: I found this warning `console.warn
    EuiButtonIcon requires aria-label or aria-labelledby to be specified because icon-only
          buttons are screen-reader-inaccessible without them.` when running unit tests, so do you think this is a valid change? If not I'll remove it.

File: public/components/DiscoverAction/SuggestAnomalyDetector.tsx
Comment (gaobinlong) [reply public/components/DiscoverAction/SuggestAnomalyDetector.tsx:421]: Thanks for pointing that, currently it will throw an error if the detector name is duplicated, but I think the behavior should be consistent with the feature anywhere, I've changed the code to reuse the existing generate detector name method.
I've uploaded the video again, please check that.

**Comment by jackiehanyang** [comment]
The code change looks good to me, just I see failed CI. Could you please fix that?

**Comment by gaobinlong** [comment]
> The code change looks good to me, just I see failed CI. Could you please fix that?

Thank you, this PR is waiting a PR in OSD core repo to be merged, I'll get all checks passed after that PR is merged.

**Comment by jackiehanyang** [comment]
@gaobinlong is this pr still tracking 2.17 release?

**Comment by gaobinlong** [comment]
> @gaobinlong is this pr still tracking 2.17 release?

Actually yes, I know today is the code freeze day, but the two PRs in other repos that this PR depends on have not been merged yet(still have some UI change), so please be lenient with this PR, thank you!

**Comment by jackiehanyang** [comment]
> > @gaobinlong is this pr still tracking 2.17 release?
> 
> Actually yes, I know today is the code freeze day, but the two PRs in other repos that this PR depends on have not been merged yet(still have some UI change), so please be lenient with this PR, thank you!

Could you link the dependency PRs here. Also please address the conflicts on this pr so that we could quickly merge in this one after the dependency PRs are merged and CI all passed. The first RC build is on 11:00am PST, Sept 5th

**Comment by gaobinlong** [comment]
> > > @gaobinlong is this pr still tracking 2.17 release?
> > 
> > 
> > Actually yes, I know today is the code freeze day, but the two PRs in other repos that this PR depends on have not been merged yet(still have some UI change), so please be lenient with this PR, thank you!
> 
> Could you link the dependency PRs here. Also please address the conflicts on this pr so that we could quickly merge in this one after the dependency PRs are merged and CI all passed. The first RC build is on 11:00am PST, Sept 5th

This PR depends on https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7936 and https://github.com/opensearch-project/dashboards-assistant/pull/265.

**Comment by gaobinlong** [comment]
@jackiehanyang thanks, I've resolved the conflicts and linked the dependency PRs yet, and also made some change for this PR, that includes:
1. Move the entry point to the query editor in Discover 2.0 page.
2. Remove the newly introduced node API `_generate_parameters`, call the node API in dashboard-assistant plugin instead.

, now the implementation is simplified, please help to take a look, thank you!

**Comment by gaobinlong** [comment]
@jackiehanyang the direct dependency [PR](https://github.com/opensearch-project/dashboards-assistant/pull/265) has been merged, so now all checks have passed yet in this PR, please help to approve and merge if no issue exists, thank you!

**Review by jackiehanyang — APPROVED** [review]

**Pull request merged**
